GooglePlacesAutocomplete.d.ts

```diff
@@ -401,19 +401,20 @@ interface GooglePlacesAutocompleteProps {
   predefinedPlaces?: Place[];
   predefinedPlacesAlwaysVisible?: boolean;
   preProcess?: (text: string) => string;
-  query: Query;
+  query: Query | Object;
   renderDescription?: (description: DescriptionRow) => string;
-  renderHeaderComponent?: () => React.ComponentType<{}>;
-  renderLeftButton?: () => React.ComponentType<{}>;
-  renderRightButton?: () => React.ComponentType<{}>;
-  renderRow?: (data: GooglePlaceData) => React.ComponentType<{}>;
+  renderHeaderComponent?: () => JSX.Element | React.ComponentType<{}>;
+  renderLeftButton?: () => JSX.Element | React.ComponentType<{}>;
+  renderRightButton?: () => JSX.Element | React.ComponentType<{}>;
+  renderRow?: (data: GooglePlaceData) => JSX.Element | React.ComponentType<{}>;
```

Comment:
```suggestion
  renderHeaderComponent?: () => JSX.Element;
  renderLeftButton?: () => JSX.Element;
  renderRightButton?: () => JSX.Element;
  renderRow?: (data: GooglePlaceData) => JSX.Element;
```

Correct me if I'm wrong, but the way these are being used in the code (snippet below), I don't think a `ComponentType` can go here.

![image](https://user-images.githubusercontent.com/10299206/98185174-7fc52200-1ee2-11eb-942a-83cfaa78724b.png)

Reply:
You are right. it is deprecated and why I added an or conditional to allow passing checks. this will need further modification but i dont fully understand why it was implemented the way it was originally and simply wished to improve it  to a point it was usable within our project. 

I can update the PR to only use JSX.Element however I was hoping to avoid breaking changes for anyone who solved for the old typedefs already.

Reply:
These typedefs were originally written a very long time ago. I am fine with it "just working" for now. Let's fix the deprecation in a separate PR, and keep the breaking changes there.